src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -483,6 +546,100 @@ class EntryStorage
         index = found ? toIndex : ArraySize(acl);
     }
 
+public:
+    static const chip::TLV::Tag kTagInUse       = chip::TLV::ContextTag(1);
+    static const chip::TLV::Tag kTagFabricIndex = chip::TLV::ContextTag(2);
+    static const chip::TLV::Tag kTagAuthMode    = chip::TLV::ContextTag(3);
+    static const chip::TLV::Tag kTagPrivilege   = chip::TLV::ContextTag(4);
+    static const chip::TLV::Tag kTagSubjects    = chip::TLV::ContextTag(5);
+    static const chip::TLV::Tag kTagTargets     = chip::TLV::ContextTag(6);
+    static const size_t kStorageBufferSize      = 192;
+
+    CHIP_ERROR Serialize(chip::PersistentStorageDelegate * storage, const char * key)
+    {
+        uint8_t buffer[kStorageBufferSize] = { 0 };
+        chip::TLV::TLVWriter writer;
+        writer.Init(buffer, sizeof(buffer));
+        chip::TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(chip::TLV::AnonymousTag, chip::TLV::TLVType::kTLVType_Structure, container));
+
+        ReturnErrorOnFailure(writer.Put(kTagInUse, mInUse));
+        ReturnErrorOnFailure(writer.Put(kTagFabricIndex, mFabricIndex));
+        ReturnErrorOnFailure(writer.Put(kTagAuthMode, mAuthMode));
+        ReturnErrorOnFailure(writer.Put(kTagPrivilege, mPrivilege));
+
+        chip::TLV::TLVType internalContainer;
+        ReturnErrorOnFailure(writer.StartContainer(kTagSubjects, chip::TLV::TLVType::kTLVType_Array, internalContainer));
+        for (size_t i = 0; i < kMaxSubjects; ++i)
+        {
+            ReturnErrorOnFailure(mSubjects[i].Serialize(writer));
```

Comment:
If these are not toplevel, why do we serialize them as structs instead of just a single value?